stablehlo/dialect/TypeInference.cpp

```diff
@@ -87,6 +87,24 @@ bool tensorsHaveSameElType(Type type1, Type type2, bool ignoreFpPrecision) {
   return tensorTy1.getElementType() == tensorTy2.getElementType();
 }
 
+LogicalResult variadicTensorsHaveSameElType(ValueRange tensors,
+                                            std::optional<Location> location) {
```

Comment:
Agreed that we need to deduplicate `tensorsHaveSameElType` and `variadicTensorsHaveSameElType`. My recommendation would be along these lines:

```C++
bool tensorsHaveSameElType(Type type1, Type type2, bool ignoreFpPrecision = true) { 
  return tensorsHaveSameElType({type1, type2}, ignoreFpPrecision);
}

bool tensorsHaveSameElType(TypeRange types, bool ignoreFpPrecision = true) { 
  // Don't emit emitOptionalError here, do that at callsites.
  // We won't be able to provide a super precise error message (that includes
  // what types mismatch), and that's fine.
  ... 
}
```